app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx

```diff
@@ -1,8 +1,46 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import Sidebar from './sidebar.jsx';
 import MenuContent from './menu_content.jsx';
+import _ from 'lodash';
 
-export default class ContentPanel extends Component {
+
+function getSignal(ownProps, state) {
+  const { route, params } = ownProps;
+
+  if (route.path === 'new') {
+    return {
+      type: params.type,
+    }
+  } else {
+    const signals = state.models.listenSignals.data;
+    const key = _.findKey(signals, { id: parseInt(params.id) });
+    const signal = signals[key];
+
+    const responses = (signal) => {
```

Comment:
Ahh yea I noticed it when i was just refreshing and realized the the props/states are only passed down when it's in the happy path.. I'll take care of that